bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/LaunchConfigFinder.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright 2024-2025 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.util;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.launchbar.core.ILaunchDescriptor;
+
+public class LaunchConfigFinder
+{
+	private final ILaunchManager launchManager;
+
+	public LaunchConfigFinder()
+	{
+		launchManager = DebugPlugin.getDefault().getLaunchManager();
+	}
+
+	public LaunchConfigFinder(ILaunchManager launchManager)
+	{
+		this.launchManager = launchManager;
+	}
+
+	public ILaunchConfiguration findAppropriateLaunchConfig(ILaunchDescriptor descriptor, String configIndentifier)
+			throws CoreException
+	{
+		IProject project = descriptor.getAdapter(IProject.class);
+		for (ILaunchConfiguration config : launchManager.getLaunchConfigurations())
+		{
+			IResource[] mappedResource = config.getMappedResources();
+			if (mappedResource != null && mappedResource.length > 0 && mappedResource[0].getProject().equals(project)
+					&& config.getType().getIdentifier().contentEquals(configIndentifier))
+			{
+				return config;
+			}
+		}
+		return null;
+	}
+
+}
```

Comment:
It's better we could rename the class to LaunchUtil or similar and then maintain all IDF Launch utiliy functions here

Reply:
Also, don't we already have ILaunchDescriptorType associated with the ILaunchDescriptor? If yes, why we are passing type again?


ILaunchDescriptorType](https://github.com/espressif/idf-eclipse-plugin/blob/e1bf806e4e74f16c7e7811397d810ceca02af326/bundles/com.espressif.idf.launch.serial.core/src/com/espressif/idf/launch/serial/core/IDFProjectLaunchDescriptor.java#L26)

Reply:
I changed the class name in the latest commit. 
> Also, don't we already have ILaunchDescriptorType associated with the ILaunchDescriptor? If yes, why we are passing type again?

we are passing active launchDescriptor from which we are getting a project and then we passing the type of config that we want to find. Here is an example:
The active launch configuration is a run type with a project "Test", but we want to find a debug config which attached to the same project "Test" to avoid creating duplicates.

Now thinking more about it, maybe it makes more sense to pass a project as an argument instead of a descriptor

Reply:
Got it. thanks